### ISSUE TITLE
[Bug]: Changing window size between image portion loads causes problems

### ISSUE THREAD

**Issue by MarisKay** [DerYeger/vue-masonry-wall#40]
### Description

Hello, i am using masonry-wall in quasar.js project. Wall is being populated in steps using q-infinite-scroll, it initially loads 25 images, then if you scroll down - another 25 images etc. All works good unless i am changing the browser's window width after my first 25 images got loaded. Then after width has been changed, i continue to scroll down , another 25 images got loaded, but bottom part suddenly uses only 2 of 3 columns leaving a huge empty space in one of the columns (which one - that's i guess depends on many other factors). Once images are loaded - i can again change the screen width and it rearranges them properly. Here's my code:

`      <q-infinite-scroll ref="infiniteScroll" @load="onFetchEntries">
        <masonry-wall v-if="entries.length>0" :items="entries" :column-width="300" :gap="16">
          <template #default="{ item }">
            <div>
              <img :src="item.picSized" style="width: 100%">
            </div>
          </template>
        </masonry-wall>
        <template #loading>
          <div class="row justify-center q-my-md">
            <q-spinner-dots color="shadeSpinner" size="40px"/>
          </div>
        </template>
      </q-infinite-scroll>`

How can this behaviour could be prevented? Thank you!

### Reproduction

Load first 25 images, change browser's window width (shrink a bit), then load extra portion of images and the problem is visible.

### Preferences

- [ ] I want to be assigned to and work on this issue myself

**Comment by MarisKay** [comment]
Sometimes you don't even have to resize the window between loads. It's enough to load another portion of images and just loaded portion at the very bottom do not utilise all columns at some point

**Comment by DerYeger** [comment]
I need an actual reproduction repository to look into this.
[The demo](https://vue-masonry-wall.yeger.eu) and all of my usages are working fine, so I'm not sure what's happening here.

**Comment by MarisKay** [comment]
Ok, i understand. Uff, i am not that good in using those codepens etc. Plus, i cannot just rip it out of my context and place in some 3rd party site. That alone will cost me days until i manage it to work :/ Can only show you the live example where it is messing up. That happens whenever q-infinite-scroll adds extra portion. So it could be that there is the answer. But what is the answer and what to look for - i have no idea, really :/

**Comment by MarisKay** [comment]
I looked into produced HTML code for the moment when i see a problem on screen. There are 3 divs with class masonry-column, counted elements with class masonry-item. In 1st column 13 elements, in second and third columns - 18 items. Does this info help somehow?

**Comment by MarisKay** [comment]
Okay, what helped me so far to eliminate the reoccurance of the problem is commenting these lines in your code in redraw function:

`    async function redraw(force = false) {
      // if (columns.value.length === columnCount() && !force) {
      //   emit("redraw-skip");
      //   return;
      // }`

Everything else stays the same and problem is long gone! Does it ring any bells? Can i safely leave these lines commented?

**Comment by DerYeger** [comment]
> Ok, i understand. Uff, i am not that good in using those codepens etc. Plus, i cannot just rip it out of my context and place in some 3rd party site. That alone will cost me days until i manage it to work :/ Can only show you the live example where it is messing up. That happens whenever q-infinite-scroll adds extra portion. So it could be that there is the answer. But what is the answer and what to look for - i have no idea, really :/

Sorry, I can't really help you if I can't reproduce the issue myself. See <https://antfu.me/posts/why-reproductions-are-required>.

> Okay, what helped me so far to eliminate the reoccurance of the problem is commenting these lines in your code in redraw function:
> 
> ` async function redraw(force = false) { // if (columns.value.length === columnCount() && !force) { // emit("redraw-skip"); // return; // }`
> 
> Everything else stays the same and problem is long gone! Does it ring any bells? Can i safely leave these lines commented?

This will redraw all elements on every resize, resulting in poor performance on low-end devices.

How are you adding new items?
Make sure to re-assign a new array instead of pushing the items to an existing array.

**Comment by MarisKay** [comment]
I also added the following console.logs in your redraw fn. My photos are added by 25 with each fetch. And no questions asked - it deisplays the correct number of items, first 25, then 50. 

items.value.length=25 (and after another portion is fetched - 50)
masonry-wall.es.js?14a3:45 force=true
masonry-wall.es.js?14a3:46 columns.value.length=3
masonry-wall.es.js?14a3:47 columnCount()=3

This is how i add new values to an array :
 entries.value = entries.value.concat(portion)

portion is current fetch of 25 items
const entries = ref([]) is where all those items are gathered

**Comment by DerYeger** [comment]
> I also added the following console.logs in your redraw fn. My photos are added by 25 with each fetch. And no questions asked - it deisplays the correct number of items, first 25, then 50.
> 
> items.value.length=25 (and after another portion is fetched - 50) masonry-wall.es.js?14a3:45 force=true masonry-wall.es.js?14a3:46 columns.value.length=3 masonry-wall.es.js?14a3:47 columnCount()=3
> 
> This is how i add new values to an array : entries.value = entries.value.concat(portion)
> 
> portion is current fetch of 25 items const entries = ref([]) is where all those items are gathered

You're adding the items incorrectly.
Try `entries.value = [...(entries.value ?? []), ...portion]`.

**Comment by MarisKay** [comment]
Hmm, what was incorrect in my approach? It still gatehered all the items in the array. 
Now i tried your way and uncommented everything back in redraw fn as it was initially. And the problem is back. So , if i add new values my way or your way - it has the same exact output - output with a problem. Can i maybe place some console.log into your code to help you gather more info? Thank you!

**Comment by DerYeger** [comment]
> Hmm, what was incorrect in my approach? It still gatehered all the items in the array. Now i tried your way and uncommented everything back in redraw fn as it was initially. And the problem is back. So , if i add new values my way or your way - it has the same exact output - output with a problem. Can i maybe place some console.log into your code to help you gather more info? Thank you!
> 
> ADD: I even tried adding elements with my removeReactivity fn which basically does this:
> 
> function removeReactivity(variable) { return JSON.parse(JSON.stringify(variable)) }
> 
> Same result also

Again, I can't really help you without some form of reproduction.

**Comment by MarisKay** [comment]
Continuing my day with testings - shows that with quasar your plugin simply start to create all weird things. As soon as i go to simple components - it all works. As soon as i uses q-infinite-scroll with q-img -> better search for alternatives

**Comment by DerYeger** [comment]
> Continuing my day with testings - shows that with quasar your plugin simply start to create all weird things. As soon as i go to simple components - it all works. As soon as i uses q-infinite-scroll with q-img -> better search for alternatives

Sounds like it may be an issue with the q-img component's height changing dynamically after the initial render.
Since this library is not the cause, I'll close the issue.

**Comment by MarisKay** [comment]
Actually no, this is not the case about the q-img loading slower or anything. It happens time to time when you add another batch of images to the array. Tested it with simple <img src=..> no quasar stuff in it. I hate to say but I spent 5 hours trying to create codepen to illustrate this, i am no spec in codepens, so it turned out in totally wasted time. In your demo example online - you add elements one by one and not images. That works. But try to add many images, like 30 images at a time with different widths heights and see how it goes. And if you create a codepen for that - i would gladly modify it to show how it is not working.
